### PR TITLE
EZP-31642: As an Editor I want be able to edit table borders

### PR DESCRIPTION
**Pull request by SerheyDolgushev**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31642](https://jira.ez.no/browse/EZP-31642)
| **Bug/Improvement**| yes
| **New feature**    | yes
| **Target version** | 2.1
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

This PR provides a new concept for custom attributes in Online Editor. In some cases, there is no need to use data attributes for them. Table border is a great example, and I guess there might be more cases.

Steps to test it:
1. Just add `border` custom attribute for tables:
    ```
    system:
        admin_group:
            fieldtypes:
                ezrichtext:
                    attributes:
                        table:
                            border: { type: number, default_value: 1 }
    ```

2. Open Online Editor and insert the table

3. Using elements path (at the bottom of the Online Editor) select a table

4. Click "Edit" icon in the table toolbar

Please note this PR depends on https://github.com/ezsystems/ezplatform-richtext/pull/150, and without schema changes, table-border edits won't be saved in the database. But it is still possible to test UI without that PR.

And some JavaScript improvement suggestions are expected here :)

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ export default class EzBtnAttributesEdit extends EzWidgetButton {
             td: this.setDefaultClassesOnTableCells,
             li: this.setDefaultClassesOnListItems,
         };
+
+        this.nativeAttributes = {
+            table: ['border']
```
Comment (dew326) [inline src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js:23]: I don't like to have this hardcoded here. 
I think it should be in the `eZ.richText` config.

File: src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js [hunk 2]
```diff
@@ -133,72 +137,94 @@ export default class EzBtnAttributesEdit extends EzWidgetButton {
...
         Object.entries(this.attributes).forEach(([attributeName, config]) => {
-            const attributeValue = block.getAttribute(`data-ezattribute-${attributeName}`);
+            const attributeValue = this.getAttributeValue(block, attributeName, config);
 
             if (attributeValue !== null) {
                 return;
             }
 
             const defaultValue = config.defaultValue;
-
```
Comment (dew326) [inline src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js:143]: Please do not remove this line. This improves readability.

File: src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js [hunk 3]
```diff
@@ -133,72 +137,94 @@ export default class EzBtnAttributesEdit extends EzWidgetButton {
...
 
+    getAttributeValue(element, name, config) {
+        const nativeAttributes = this.nativeAttributes[element.getName()] || [];
+
+        if (nativeAttributes.includes(name)) {
+            return element.getAttribute(name);
+        }
+
+        let value = element.getAttribute(`data-ezattribute-${name}`);
+
```
Comment (dew326) [inline src/bundle/Resources/public/js/OnlineEditor/buttons/ez-btn-attributes-edit.js:199]:
```suggestion
```

**Review by dew326 — CHANGES_REQUESTED** [review]

**Comment by SerheyDolgushev** [comment]
@dew326 please check all the requested changes in https://github.com/ezsystems/ezplatform-richtext/pull/151/commits/e57e930945b20223b4aa67f9792a694eef9d0a9e

**Review by dew326 — APPROVED** [review]
+1 for the frontend part

**Review by GrabowskiM — APPROVED** [review]

**Review by lucasOsti — APPROVED** [review]

**Comment by micszo** [comment]
@SerheyDolgushev I believe this should be rebased against 2.1, v3.0 reached EOL in August. (and in fact v3.1 will be past EOM tomorrow, https://support.ibexa.co/Public/Service-Life#)

**Comment by SerheyDolgushev** [comment]
@micszo rebased onto `2.1`.

**Review by katarzynazawada — APPROVED** [review]
Tested together with https://github.com/ezsystems/ezplatform-richtext/pull/150

**Comment by lserwatka** [comment]
@alongosz could you merge it up?

**Comment by alongosz** [comment]
Merged up #150 and #151 to master via e2bdfdc47.

**Comment by alongosz** [comment]
Thank you @SerheyDolgushev :tada:

**Pull request merged**